src/main/kotlin/org/jetbrains/research/ml/dataset/anonymizer/Anonymizer.kt

```diff
@@ -27,20 +29,47 @@ abstract class Anonymizer(private val tmpDataPath: String) {
     protected abstract val project: Project
     private var counter: Int = 0
 
-    protected fun anonymize(code: String): PsiElement {
-        val file = createFile("$tmpDataPath/tmp_${counter}${language.extension.value}", code)
-        val virtualFile = LocalFileSystem.getInstance().refreshAndFindFileByIoFile(file)
-            ?: error("The virtual file ${file.path} was not created")
-        val psi = ApplicationManager.getApplication().runReadAction<PsiElement> {
-            PsiManager.getInstance(project).findFile(virtualFile) as PsiElement
+    inner class PsiCreator(code: String) {
+        private val file = createFile("$tmpDataPath/tmp_${counter}${language.extension.value}", code)
+        val psi = createPsi()
+
+        private fun createPsi(): PsiElement {
+            val virtualFile = LocalFileSystem.getInstance().refreshAndFindFileByIoFile(file)
+                ?: error("The virtual file ${file.path} was not created")
+            return ApplicationManager.getApplication().runReadAction<PsiElement> {
+                PsiManager.getInstance(project).findFile(virtualFile) as PsiElement
+            }
         }
+
+        fun deleteFile() {
+            ApplicationManager.getApplication().invokeAndWait {
+                ApplicationManager.getApplication().runWriteAction {
+                    FileUtil.delete(file)
+                }
+            }
+            LocalFileSystem.getInstance().refreshAndFindFileByIoFile(file)
+        }
+    }
+
+    private fun anonymize(code: String): PsiElement {
+        val psiCreator = PsiCreator(code)
+        anonymize(psiCreator.psi)
+        psiCreator.deleteFile()
+        return psiCreator.psi
+    }
+
+    // Only for tests
+    private fun anonymize(code: String, fixture: CodeInsightTestFixture): PsiElement {
+        val fileName = "dummy." + PythonFileType.INSTANCE.defaultExtension
+        val psiFile = fixture.configureByText(fileName, code)
+        anonymize(psiFile)
+        return psiFile
```

Comment:
Cannot we move it to tests as an extension function?

Reply:
No, because anonymize is an inner class function (not static)